src/sentry/models/event.py

```diff
@@ -353,11 +354,53 @@ def prev_event(self):
             group_id=self.group_id,
         ).exclude(id=self.id).order_by('-datetime')[0:5]
 
-        events = [e for e in events if e.datetime == self.datetime and e.id < self.id
-                  or e.datetime < self.datetime]
+        events = [e for e in events if e.datetime == self.datetime and e.id < self.id or
+                  e.datetime < self.datetime]
         events.sort(key=EVENT_ORDERING_KEY, reverse=True)
         return events[0] if events else None
 
+    @classmethod
+    def get_event(cls, id_or_event_id, project_id=None):
+        """
+        Get an Event by either its id primary key or its hex event_id.
+
+        Will automatically try to infer the type of id, and grab the correct
+        event.  If the provided id is a hex event_id, the project_id must also
+        be provided to disambiguate it.
+
+        Returns None if the event cannot be found under either scheme.
+        """
+        # TODO (alexh) instrument this to report any times we are still trying
+        # to get events by id.
+        # TODO (alexh) deprecate lookup by id so we can move to snuba.
+
+        event = None
+        if id_or_event_id.isdigit():
+            # If its a numeric string, check if it's an event Primary Key first
+            try:
+                if project_id is None:
+                    event = cls.objects.get(
+                        id=id_or_event_id,
```

Comment:
Doesn't this allow any user to fetch any event ID? Don't we *have* to scope by Project or Organization?

Reply:
I think the permissions checks are more or less left to the API handlers, but I don't know enough to be 100% confident that they all do the correct checks.

At least, as far as I can tell, this isn't any more or less permissive than the queries it's replacing.

Reply:
I could require the project_id, but then `event_apple_crash_report.py` and `event_details.py` would need to change to provide it, and those don't seem to actually have any way to get the project, so that might not be possible.

Reply:
Yeah, it's not more permissive it just might hide the fact that the user is calling with an event and not a project, where an ORM call is more obvious (to me).

Dumb idea: What if you made `project_id` required in the signature and force-passed a `project_id=None` for those cases?

Reply:
done